### PR TITLE
Update vacancy API to allow listing without hire stream filter

### LINKED ISSUE #493: Update vacancy API to allow listing without hire stream filter
Change hire stream filter to allow multiple values or an empty array

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #493

## PR includes

-   [x] Feature

### CODE REVIEW

File: src/modules/vacancyMethods.ts [hunk 1]
```diff
@@ -27,18 +39,53 @@ export const vacancyMethods = {
...
+            where,
+            include: { group: { include: { supervisor: true } }, hr: true, hiringManager: true },
+            take: take || defaultTake,
+            skip: skip || 0,
+        });
     },
 
-    getById: (id: string) => prisma.vacancy.findFirstOrThrow({ where: { id }, include: { group: true, user: true } }),
+    getById: (id: string) =>
+        prisma.vacancy.findFirstOrThrow({
```
Comment (tsumo) [inline src/modules/vacancyMethods.ts:55]: better to throw TRPCError 404 in cases like this

File: src/modules/vacancyMethods.ts [hunk 2]
```diff
@@ -27,18 +39,53 @@ export const vacancyMethods = {
...
+
+        if (restInput.status) {
+            const previousVacancy = await prisma.vacancy.findFirstOrThrow({
+                where: { id },
+            });
+
+            if (previousVacancy.status === VacancyStatus.ACTIVE && restInput.status !== VacancyStatus.ACTIVE) {
+                const today = new Date();
+                const activeSince = previousVacancy.activeSince || new Date();
+                data.timeAtWork = previousVacancy.timeAtWork || 0 + today.getTime() - activeSince.getTime();
```
Comment (tsumo) [inline src/modules/vacancyMethods.ts:72]: Is this time duration in milliseconds?

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/modules/vacancyMethods.ts (hunk 2 above)
Comment (nyet-ty) [reply src/modules/vacancyMethods.ts:72]: yes, it is

File: src/trpc/router/restRouter.ts [hunk 3]
```diff
@@ -152,7 +153,6 @@ export const restRouter = router({
                                 name: z.string(),
                             }),
                         ),
-                        percentage: z.number().nullable(),
```
[the remark this answers is not in the document]
Comment (nyet-ty) [reply src/trpc/router/restRouter.ts:155]: sorry...

**Review by 9teen90nine — APPROVED** [review]

**Review by tsumo — APPROVED** [review]

**Pull request merged**